presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -271,23 +272,21 @@ else if (firstDependencies.stream().allMatch(right.canResolvePredicate()) && sec
             }
 
             // Add projections for join criteria
-            leftPlanBuilder = appendProjections(leftPlanBuilder, leftExpressions);
-            rightPlanBuilder = appendProjections(rightPlanBuilder, rightExpressions);
+            leftPlanBuilder = appendProjections(leftPlanBuilder, leftComparisonExpressions);
+            rightPlanBuilder = appendProjections(rightPlanBuilder, rightComparisonExpressions);
 
-            List<Expression> postInnerJoinComparisons = new ArrayList<>();
-            for (int i = 0; i < comparisonTypes.size(); i++) {
-                Symbol leftSymbol = leftPlanBuilder.translate(leftExpressions.get(i));
-                Symbol rightSymbol = rightPlanBuilder.translate(rightExpressions.get(i));
+            for (int i = 0; i < leftComparisonExpressions.size(); i++) {
+                Symbol leftSymbol = leftPlanBuilder.translate(leftComparisonExpressions.get(i));
+                Symbol rightSymbol = rightPlanBuilder.translate(rightComparisonExpressions.get(i));
 
-                if (comparisonTypes.get(i) == ComparisonExpression.Type.EQUAL) {
+                if (joinConditionComparisonTypes.get(i) == ComparisonExpression.Type.EQUAL) {
                     equiClauses.add(new JoinNode.EquiJoinClause(leftSymbol, rightSymbol));
```

Comment:
Seems like there should be a `continue` here. (You can put that in a separate commit though, since it's a pre-existing bug)